pgml-extension/src/bindings/transformers.py

```diff
@@ -39,19 +39,26 @@
 
 __cache_transformer_by_model_id = {}
 __cache_sentence_transformer_by_name = {}
+__cache_transform_pipeline_model_by_name = {}
 
 class NumpyJSONEncoder(json.JSONEncoder):
     def default(self, obj):
         if isinstance(obj, np.float32):
             return float(obj)
         return super().default(obj)
 
-def transform(task, args, inputs):
+def transform(task, args, inputs, cache_model):
     task = json.loads(task)
     args = json.loads(args)
     inputs = json.loads(inputs)
 
-    pipe = transformers.pipeline(**task)
+    model = task.get("model")
```

Comment:
I think there may be different pipelines with the same model, to handle different tasks. We may need to use the full task param for caching. Something like:

```
    if cache:
        key = ','.join([str(key) + ':' + str(value) for (key, value) in sorted(task.items())])
        if key not in __cache_transformer_by_task:
            __cache_transformer_by_task[key] = transformers.pipeline(**task)
        pipe = __cache_transformer_by_task[key]
    else:
        pipe = transformers.pipeline(**task)
```

Reply:
I see, good point. I'll think some more about the parameters of .pipeline() and push an update.

Reply:
Okay, I pretty much copied your code verbatim. I also like the idea in there where you have to pass the `cache` parameter also to USE the cached model. My initial version always used the cached version if it was available whether or not the cache flag was passed in or not.